pkg/agent/ippool/event.go

```diff
@@ -117,8 +114,23 @@ func (e *EventHandler) EventListener() (err error) {
 
 	controller := NewController(queue, indexer, informer, e.poolRef, e.dhcpAllocator, e.poolCache)
 	stop := make(chan struct{})
-	defer close(stop)
+
 	go controller.Run(1, stop)
 
-	select {}
+	<-stopCh
+	controller.Stop(stop)
+
+	logrus.Info("(eventhandler.Run) IPPool event listener terminated")
+}
+
+func (e *EventHandler) stop(stopCh chan struct{}) {
+	logrus.Info("(eventhandler.Stop) stopping IPPool event listener")
+	close(stopCh)
+}
+
+func Cleanup(ctx context.Context, e *EventHandler, stopCh chan struct{}) {
+	go func() {
+		<-ctx.Done()
+		e.stop(stopCh)
+	}()
```

Comment:
In this `Cleanup`, you will close `stopCh`. After closing `stopCh`, you will call `contorller.Stop(stop)`.

Reply:
The `stop` channel passed in as an argument of `controller.Stop()` is the one declared in the `EventListener()` scope, not the one declared in the `Agent.Run()` scope. They are different ones. Maybe I can rename them to avoid confusion.

Reply:
oh, I see. No problem here. Thanks!

Reply:
It's okay not to rename it.